### ISSUE TITLE
Restore works with config file but not with command line options

### ISSUE THREAD

**Issue by bradnicholson** [pgbackrest/pgbackrest#502]
1. pgBackRest version: 2.00
2. PostgreSQL version: 9.6.7

I have a case where restore from s3 fails when I provide command line arguments.  But if I put the same arguments into a config file, it works.  I've double checked the obvious on the redacted info in my test (ensuring the keys and other values are the same command line vs file).

Command line.  
```
/usr/bin/pgbackrest --log-level-console=warn --stanza=test --db-path=/tmp/postgres --repo-type=s3 --repo-path=/my_pg_backup --repo-s3-endpoint=my_endpoint --repo-s3-region=my-region --repo-s3-key=my-key --repo-s3-bucket=my-bucket --repo-s3-key-secret=my-key-secret restore
```
Fails with:
`ERROR [025]: : raised from local-1 process: option repo1-s3-key is required`

If I provide the same data via this config file, it restores properly.

```
[global]
log-level-console=warn
repo-type=s3
repo-path=/my_pg_backup
repo-s3-endpoint=my_endpoint
repo-s3-region=my-region
repo-s3-key=my-key
repo-s3-bucket=my-bucket
repo-s3-key-secret=my-key-secret

[test]
db-path=/tmp/postgres
```

Called with:
```
pgbackrest --config=/tmp/restore.conf --stanza=tets restore
```

works.

**Comment by dwsteele** [comment]
Hi Brad,

This is intentional, though it looks like we need a better error.  pgBackRest will refuse to pass secrets to local/remote processes on the command-line for security.  The main reason is that they will show up in ps and that doesn't seem like a good thing.

I think it should just error as soon as it sees a secret option specified on the command-line.  Thoughts?

**Comment by bradnicholson** [comment]
Makes sense.

Agree - erroring right away with a descriptive message would make sense.

Also, the docs read like this is possible, so I'd suggest correcting/clarifying it there as well.  For example:

https://pgbackrest.org/command.html#command-archive-push/category-repository/option-repo-s3-key-secret

**Comment by dwsteele** [comment]
Good point about the docs.  We'll update those as well.

**Comment by dwsteele** [comment]
Fixed at 8a1ce42c30e7e9ac79b1f.  It will go out with the 2.01 release.

Thanks for the report!
